src/sources/internal_logs.rs

```diff
@@ -341,6 +344,43 @@ mod tests {
         rx
     }
 
+    // NOTE: This test requires #[serial] because it directly interacts with global tracing state.
+    // This is a pre-existing limitation around tracing initialization in tests.
+    #[tokio::test]
+    #[serial]
+    async fn repeated_logs_are_not_rate_limited() {
+        trace::init(false, false, "info", 10);
+        trace::reset_early_buffer();
+
+        let rx = start_source().await;
+
+        // Generate 20 identical log messages with the same component_id
+        for _ in 0..20 {
+            info!(component_id = "test", "Repeated test message");
+        }
+
+        sleep(Duration::from_millis(50)).await;
```

Comment:
Not the biggest fan of the arbitrary sleep here. Can this be removed or does the test become flaky?